HealthcareSystem/PatientWebApp/wwwroot/js/patient_examinations.js

```diff
@@ -618,10 +644,21 @@ function third_step_next() {
                 $('form#schedule').find(":submit").prop('disabled', false);
             }
             else {
-                for (let a of appointments) {
-                    let appointment = $('<option value="' + i + '">' + a.startTime + '</option>');
-                    $('#free_appointments').append(appointment);
-                    i = i + 1;
+                for (let j = 0; j < appointments.length; j++) {
+                    for (let k = 0; k < appointmentStartTimes.length; k++) {
+                        if (appointments[j].startTime == appointmentStartTimes[k]) {
+                            multipleTimes = true;
+                            break;
+                        }
+                    }
+                    if (multipleTimes == false) {
+                        let appointment = $('<option value="' + i + '">' + appointments[j].startTime + '</option>');
+                        appointmentStartTimes.push(appointments[j].startTime);
+                        $('#free_appointments').append(appointment);
+                        i = i + 1;
+                    } else {
+                        multipleTimes = false;
+                    }
```

Comment:
You no longer have all the appointments in the combo box, but you are using a list with all the appointments when you try to schedule. Therefore, you will send an AJAX that tries to schedule the wrong appointment:
![image](https://user-images.githubusercontent.com/57844740/105644179-52aa5f00-5e94-11eb-8c02-730574680c7d.png)
![image](https://user-images.githubusercontent.com/57844740/105644189-65249880-5e94-11eb-8b2e-ff231c86c005.png)

Reply:
I returned the previous version (develop version) of this js file in order to merge this branch into develop as soon as posible. We will take care of this on another branch.

Reply:
The issue was fixed on this branch instead since returning to the previous version was difficult (the changes related to the examination filtering were mixed in with event related changes). The issue appears to be completely fixed as of 3596344bef37d835456649237e25d4994465dd14.